From e07b73b5113477a92f1e7f6917e31c954b3ccaf3 Mon Sep 17 00:00:00 2001
From: Anirudh Ghayal <aghayal@codeaurora.org>
Date: Mon, 30 Mar 2020 16:50:21 +0530
Subject: [PATCH] power: smb5-lib: Query FLASH_TRIGGER only for PMI632

The FLASH_TRIGGER property is supported only on PMI632, add
a check for it.

Change-Id: I269b0be18b56d56c0ab0e68ee1d7194d09824dd3
Signed-off-by: Anirudh Ghayal <aghayal@codeaurora.org>
---
 drivers/power/supply/qcom/qpnp-smb5.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/power/supply/qcom/qpnp-smb5.c b/drivers/power/supply/qcom/qpnp-smb5.c
index 1546962d7f50..199c1f375882 100644
--- a/drivers/power/supply/qcom/qpnp-smb5.c
+++ b/drivers/power/supply/qcom/qpnp-smb5.c
@@ -1344,7 +1344,9 @@ static int smb5_usb_main_get_prop(struct power_supply *psy,
 		val->intval = chg->flash_active;
 		break;
 	case POWER_SUPPLY_PROP_FLASH_TRIGGER:
-		rc = schgm_flash_get_vreg_ok(chg, &val->intval);
+		val->intval = 0;
+		if (chg->chg_param.smb_version == PMI632_SUBTYPE)
+			rc = schgm_flash_get_vreg_ok(chg, &val->intval);
 		break;
 	case POWER_SUPPLY_PROP_TOGGLE_STAT:
 		val->intval = 0;
